File: Languages/Java/Day022.java
// Languages - Java - Day 22
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that removes all duplicate characters from a string while preserving the original order.


import java.util.LinkedHashSet;
import java.util.Scanner;

public class Day022 {
    // Removes duplicate characters while preserving the original order.
    public static String removeDuplicates(String input) {
        if (input == null || input.isEmpty()) {
            return input;
        }
        LinkedHashSet<Character> set = new LinkedHashSet<>();
        StringBuilder result = new StringBuilder();
        for (char c : input.toCharArray()) {
            if (set.add(c)) {
                result.append(c);
            }
        }
        return result.toString();
    }

    // Get user input and display the result
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter a string:");
        String input = scanner.nextLine();
        if (input.isEmpty()) {
            System.out.println("Input cannot be empty.");
            scanner.close();
            return;
        }
        String result = removeDuplicates(input);
        System.out.println("String after removing duplicates: " + result);
        scanner.close();
    }
}

File: Languages/Kotlin/Day022.kt
// Languages - Kotlin - Day 22
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that removes all duplicate characters from a string while preserving the original order.


fun removeDuplicates(input: String): String {
    if (input.isEmpty()) {
        return "Input cannot be empty."
    }
    val seen = mutableSetOf<Char>()
    val result = StringBuilder()
    for (char in input) {
        if (seen.add(char)) {
            result.append(char)
        }
    }
    return result.toString()
}

// Get user input and display the result
fun main() {
    println("Enter a string:")
    val input = readLine()?.trim() ?: ""
    if (input.isEmpty()) {
        println("Input cannot be empty.")
        return
    }
    val result = removeDuplicates(input)
    println("String after removing duplicates: $result")
}
